src/script.ts

```diff
@@ -127,11 +157,13 @@ class Script {
         });
       } else if (bitcom === "1PuQa7K62MiKCtssSLKy1kh56WWU7MtUR5") {
         const type = cell.shift();
-        const map = {};
+        const map: Record<string, any> = {};
         while (cell.length > 0) {
-          const key = cell.shift().toString();
-          const value = cell.shift();
-          map[key] = value ? value.toString() : "";
+          const key = cell.shift()?.toString();
+          if (key) {
+            const value = cell.shift();
+            map[key] = value ? value.toString() : "";
+          }
```

Comment:
Using `shift` can never guarantee a value is returned, is this `if (key)` check OK here?